src/Atoms/UserProfileData/UserProfileData.js

```diff
@@ -0,0 +1,82 @@
+import propTypes from 'prop-types';
+import { useContext, useEffect, useState } from 'react';
+
+import UserProfileDataStyled from './UserProfileDataStyled';
+import participants from '../../Data/MongoDBMock/participants';
+import { UserDataContext } from '../../Context/UserDataContext';
+
+// import { useLocation } from 'react-router-dom';
+
+const initialData = {
+  address: {
+    country: 'Polska',
+    city: 'Sfornegacie',
+    street: 'ul.Pszczelna',
+    numberOfHouse: '27/5',
+    postalCode: '50-124',
+  },
+};
+
+const UserProfileData = () => {
+  const { state } = useContext(UserDataContext);
+  const { userId, userName, userSurname, isAuthenticated } = state;
+  const [userObject, setUserObject] = useState(initialData);
+
+  if (!isAuthenticated) {
+    throw new Error('Your not allowed to be here!');
+  }
+
+  // mock for checking authentication and if userId is in database
+
+  // const { pathname } = useLocation();
+
+  // if (!isAuthenticated) {
+  //   try {
+  //     const userId = pathname.split('/').pop();
+
+  //     const userObject = participants.find(
+  //       (participant) => participant.participantId === userId,
+  //     );
+  //     dispatch({ type: 'LOG_IN' });
+  //     setUserObject(userObject);
+  //   } catch (err) {
+  //     throw new Error('Your not allowed to be here!');
+  //   }
+  // }
+
+  const { address } = userObject;
+  const { street, numberOfHouse, city, postalCode } = address;
+
+  // mock for fetching data from database and checking if response has data
+
+  useEffect(() => {
+    const userObject = participants.find(
+      (participant) => participant.participantId === userId,
+    );
+    if (userObject === undefined) {
+      throw new Error('Fetch was unsuccessful');
+    } else {
+      setUserObject(userObject);
+    }
+  }, []);
+
+  return (
+    <UserProfileDataStyled>
+      <h3>{`${userName} ${userSurname}`}</h3>
+      <p>{`${street} ${numberOfHouse}`}</p>
+      <p>{`${postalCode} ${city}`}</p>
```

Comment:
If you won't be able to destructure - you can always create a function that will return the values in the desired format and call it here